GiniVision/Classes/Core/Gallery/ImagePickerViewController.swift

```diff
@@ -0,0 +1,137 @@
+//
+//  ImagePickerViewController.swift
+//  GiniVision
+//
+//  Created by Enrique del Pozo Gómez on 2/26/18.
+//
+
+import Foundation
+import Photos
+
+protocol ImagePickerViewControllerDelegate: class {
+    func imagePicker(_ viewController: ImagePickerViewController, didSelectAsset asset: Asset)
+    func imagePicker(_ viewController: ImagePickerViewController, didDeselectAsset  asset: Asset)
+}
+
+final class ImagePickerViewController: UIViewController {
+    
+    let currentAlbum: Album
+    weak var delegate: ImagePickerViewControllerDelegate?
+    fileprivate let galleryManager: GalleryManagerProtocol
+    private var isInitialized: Bool = false
+    private let multipleSelectionLimit: Int = 10
+    
+    // MARK: - Views
+    
+    lazy var collectionView: UICollectionView = {
+        let collectionLayout = UICollectionViewFlowLayout()
+        collectionLayout.minimumLineSpacing = 1
+        collectionLayout.minimumInteritemSpacing = 1
+        
+        let collectionView = UICollectionView(frame: .zero, collectionViewLayout: collectionLayout)
+        collectionView.translatesAutoresizingMaskIntoConstraints = false
+        collectionView.dataSource = self
+        collectionView.delegate = self
+        collectionView.backgroundColor = .clear
+        collectionView.allowsMultipleSelection = true
+        collectionView.register(ImagePickerCollectionViewCell.self,
+                                forCellWithReuseIdentifier: ImagePickerCollectionViewCell.identifier)
+        return collectionView
+    }()
+    
+    // MARK: - Initializers
+    
+    init(album: Album,
+         galleryManager: GalleryManagerProtocol,
+         giniConfiguration: GiniConfiguration) {
+        self.galleryManager = galleryManager
+        self.currentAlbum = album
+        super.init(nibName: nil, bundle: nil)
+    }
+    
+    required init?(coder aDecoder: NSCoder) {
+        fatalError("init(giniConfiguration:) has not been implemented")
+    }
+    
+    // MARK: - UIViewController
+    
+    override func loadView() {
+        super.loadView()
+        
+        title = currentAlbum.title
+        view.backgroundColor = .white
+        view.addSubview(collectionView)
+        
+        Constraints.pin(view: collectionView, toSuperView: view)
+    }
+    
+    override func viewDidLayoutSubviews() {
+        super.viewDidLayoutSubviews()
+        scrollToBottomOnStartup()
+    }
+    
+    // MARK: - Others
+ 
+    fileprivate func scrollToBottomOnStartup() {
+        // This tweak is needed to fix an issue with the UICollectionView. UICollectionView doesn't
+        // scroll to the bottom on `viewWillAppear`, which is right after `viewDidLayoutSubviews`.
+        // Since this method can be called several times during the lifecycle, there should be
+        // a one-time scrolling before the view appears for the first time.
+        if !isInitialized {
```

Comment:
what is you used `Dispatch.once`or whatever the name was?

Reply:
`Dispatch
The free function dispatch_once is no longer available in Swift. In Swift, you can use lazily initialized globals or static properties and get the same thread-safety and called-once guarantees as dispatch_once provided.`